get_info.php

```diff
@@ -201,56 +228,98 @@ function get_list_of_surveys(
-        $result = [];
-        //specific survey data for each survey requested
-        for ($i = 0; $i < sizeof($survey_choices); $i++) {
-            $survey_data = $survey_choices[$i];
-            $survey_data["questions"] = [];
+    } elseif ($survey_id) {
+        //invalid survey_id check
+        $valid_surveys = filter_surveys(
+            $survey_choices,
+            $bind_variables[':survey_id'],
+            $is_instance
+        );
 
-            //no number locked by course when viewing admin survey
-            if (!isset($survey_data["number_locked_by_course"])) {
-                $survey_data["number_locked_by_course"] = null;
-            }
+        //get all surveys selected
+        $survey_data = get_query_result(gen_query_get_survey_data(), [
+            ":survey_id" => $valid_surveys
+        ]);
 
-            $responses = [];
+        //foreach survey_data combine the choices to create one choices attribute
+        foreach ($survey_data as $index => $value) {
+            // get 3 sets of survey choices
+
+            $choices = get_query_result(gen_query_get_survey_choices(), [
+                ":survey_id" => $value["survey_id"]
+            ])[0];
+            $all_choices = [
+                $choices["dept_choices_id"],
+                $choices["course_choices_id"],
+                $choices["ta_choices_id"]
+            ];
+            $survey_data[$index]["questions"] = [];
+
+            //get survey responses if the survey results are requested
+            $responses = null;
             if ($is_instance) {
                 //get list of responses when survey_results is requested
                 $responses = get_query_result(gen_query_survey_responses(), [
                     ":survey_id" => $bind_variables[":survey_id"]
                 ]);
             }
+            //default set of choices + 3 set of choices for the provided survey
+            $choice_set = $default_choices;
+            for ($i = 0; $i < 3; $i++) {
+                $choices = get_query_result(gen_query_get_choices(), [
+                    ":choices_id" => $all_choices[$i]
+                ]);
 
-            // set response and position for each selected question
-            for ($j = 1; $j <= 6; $j++) {
-                $question = $list_of_quesitons[$survey_data["choice" . $j] - 1];
-                $question["responses"] = isset(
-                    $responses[$survey_data["choice" . $j] - 1]
-                )
-                    ? $responses[$survey_data["choice" . $j] - 1]["answers"]
+                //if choice is not empty
+                if (!sizeof($choices) == 0) {
+                    //select the choices that the role can override
+                    //if a choice is not null:
+                    //dept will override all 6 choices
+                    //then course will override the last 4 choices
+                    //then ta will override the last two choices
+                    //example:
```

Comment:
This is what I thought. Don't put this complicated logic here. Assume the data is correct. That is, assume the first 4 questions in `ta_choices` are `null` and the first two questions in `course_choices` are `null`.